instance/tasks.py

```diff
@@ -60,3 +63,51 @@ def spawn_appserver(instance_ref_id, mark_active_on_success=False, num_attempts=
                 # finish and replace the second as the active server. We are not really worried about that for now.
                 instance.set_appserver_active(appserver_id)
             break
+
+
+@db_task()
+def shut_down_instances():
+    """
+    Shut down instances whose PRs got merged (more than) one week ago.
+
+    Terminate all app servers belonging to these instances, and disable monitoring.
+    """
+    for instance in OpenEdXInstance.objects.filter(watchedpullrequest__isnull=False):
+        pr = github.get_pr_info_by_number(
+            instance.watchedpullrequest.target_fork_name,
+            instance.watchedpullrequest.github_pr_number
+        )
+        if pr['state'] == 'closed':
+            closed_at = github.parse_date(pr['closed_at'])
+            now = datetime.now()
+            if sufficient_time_passed(closed_at, now, 7):
+                instance.disable_monitoring()
+                for appserver in instance.appserver_set.all():
+                    appserver.terminate_vm()
```

Comment:
@itsjeyd Could these three lines be moved to a `instance.shutdown()` method, to keep this logic internal? This task doesn't need to know how the shutdown happens.